metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py

```diff
@@ -58,11 +60,47 @@
 VIEW_LINEAGE = "view_lineage"
 
 
+class UpstreamColumnNode(TypedDict):
+    object_name: str
+    object_domain: str
+    column_name: str
+
+
+class ColumnUpstreamJob(TypedDict):
+    column_upstreams: List[UpstreamColumnNode]
+    query_id: str
+
+
+class ColumnUpstreamLineage(TypedDict):
+    column_name: str
+    upstreams: List[ColumnUpstreamJob]
+
+
+class UpstreamTableNode(TypedDict):
+    upstream_object_domain: str
+    upstream_object_name: str
+    query_id: str
+
+
+class Query(TypedDict):
+    query_id: str
+    query_text: str
+    start_time: str
+
+
+class UpstreamLineageEdge(TypedDict):
```

Comment:
if we made these all pydantic objects, we could just call .parse_obj and not manually map everything

the JSON stuff could also get fixed up by a validator (similar to what we do here https://github.com/acryldata/acryl-executor/blob/5a3bbf92386a1af68661c5d0f55bd1bb437ea578/src/acryl/executor/execution/sub_process_task_common.py#L151)

Reply:
Done